source/fundamentals/connection.txt

```diff
@@ -0,0 +1,510 @@
+   :atlas:`Atlas driver connection guide <driver-connection?jmp=docs_driver_java>`
+   to retrieve your connection string.
+
+The next part of the connection string contains your username and password
+if you are using password-based authentication. Replace the value of ``user``
+with your username and ``pass`` with your password. If you are using an
+authentication mechanism that does not require a username and password, omit
+this part of the connection URI.
+
+The next part of the connection string specifies the hostname or IP address and
+port of your MongoDB instance. In the example below, we use ``sample-hostname``
+as the hostname and ``27017`` as the port. Replace these values to point to
+your MongoDB instance.
+
+The last part of the connection string contains connection and authentication
+options as parameters. In the example below, we set two connection options:
+``retryWrites=true`` and ``w=majority``. For more information on connection
+options, skip to the :ref:`connection options <connection-options>` section.
+
+The code below shows how you can use the sample connection URI in a client to
+connect to MongoDB.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/srv.java
+   :language: java
+
+.. _jndi:
+
+Java Naming and Directory Interface (JNDI)
+------------------------------------------
+
+MongoClientFactory includes a `JNDI <http://docs.oracle.com/javase/8/docs/technotes/guides/jndi/index.html>`__
+``ObjectFactory`` implementation that returns ``MongoClient`` instances
+based on a :ref:`connection URI <connection-uri>`. Follow the guides
+below to configure your application to connect using a JNDI DataSource.
+
+.. _jndi-wildfly:
+
+Wildfly (formerly JBoss)
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. In a `Wildfly <http://wildfly.org/>`__ installation, create a new module
+   for MongoDB at `modules/system/layers/base/org/mongodb/main`. Copy the
+   ``mongo-java-driver.jar`` jar file into the module. Add the following
+   ``module.xml`` file into the module:
+
+   .. literalinclude:: /includes/fundamentals/code-snippets/wildfly-module-4.0.0.xml
+      :language: xml
+
+
+#. Add a binding to the naming subsystem configuration that references the
+   above module, the ``MongoClientFactory`` class, and the
+   :ref:`connection string <connection-uri>` for the MongoDB cluster.
+
+   .. code-block:: xml
+
+      <subsystem xmlns="urn:jboss:domain:naming:2.0">
+         <bindings>
+            <object-factory name="java:global/MyMongoClient" module="org.mongodb" class="com.mongodb.client.MongoClientFactory">
+               <environment>
+                  <property name="connectionString" value="mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority"/>
+               </environment>
+            </object-factory>
+         </bindings>
+         <remote-naming/>
+      </subsystem>
+
+   .. note::
+
+      Replace the placeholder connection value URI here with a value that
```

Comment:
Nitpick: rather than "here", it could be useful to specify "in the property tag" since we don't have emphasize-lines to give them a direct cue.